src/Components/Web/src/Forms/InputFile/IBrowserFile.cs

```diff
@@ -38,7 +38,20 @@ public interface IBrowserFile
         /// <summary>
         /// Opens the stream for reading the uploaded file.
         /// </summary>
+        /// <param name="maxAllowedSize">
+        /// The maximum size that can be read from the Stream. Defaults to 500kb.
+        /// <para>
+        /// Calling <see cref="OpenReadStream(long, CancellationToken)"/>
+        /// will throw if the uploaded file size, as specified by <see cref="Size"/> is larger than
+        /// <paramref name="maxAllowedSize"/>. By default, uploading a file larger than 500kb will result in an exception.
+        /// </para>
+        /// <para>
+        /// When specified, it is important to choose a value based on the largest size the calling code is expected to handle.
+        /// Specifying an arbitrary large value may by a vector for resource exhaustion attacks in particular in Blazor Server.
```

Comment:
Suggest avoiding the term "vector" as it's a bit jargony outside security circles. Here's an attempt to phrase it more approachably for beginners, but please feel free to use your own description if you prefer!

```suggestion
        /// It is valuable to choose a size limit that corresponds to your use case. If you allow excessively large files, this
        /// may result in excessive consumption of memory or disk/database space, depending on what your code does
        /// with the supplied Stream.
        /// </para>
        /// <para>
        /// For Blazor Server in particular, beware of reading the entire stream into a memory buffer unless you have
        /// passed a suitably low size limit, since you will be consuming that memory on the server.
```